src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -170,9 +170,82 @@ private static bool ApplyChangesFromTC35(JObject sarifLog)
 
             modifiedLog |= RenameFixChangesToFixArtifactChanges(sarifLog);
 
+            modifiedLog |= RenameArtifactRolesEnums(sarifLog);
+
             return modifiedLog;
         }
 
+        private static bool RenameArtifactRolesEnums(JObject sarifLog)
+        {
+            string[] artifactRolesPathsToUpdate =
+            {
+                "inlineExternalProperties[].artifacts[]",
+                "runs[].artifacts[]"
+            };
+
+            bool actionOnLeafNode(JObject artifact)
+            {
+                if (artifact["roles"] is JArray roles)
+                {
+                    bool isModified = false;
+
+                    foreach(JValue role in roles)
+                    {
+                        switch (role.Value as string)
+                        {
+                            case "unmodifiedFile":
+                                {
+                                    role.Value = "unmodified";
+                                    isModified = true;
+                                    break;
+                                }
+                            case "modifiedFile":
+                                {
+                                    role.Value = "modified";
+                                    isModified = true;
+                                    break;
+                                }
+                            case "addedFile":
+                                {
+                                    role.Value = "added";
+                                    isModified = true;
+                                    break;
+                                }
+                            case "deletedFile":
+                                {
```

Comment:
>{ [](start = 32, length = 1)

your braces aren't indented to our style, you need to change your VS settings. these should be left aligned with the 'case' keyword. otherwise the contents of the case is effectively tabbed twice, which is a lot of right shifting

Reply:
i've approved but feel free to fix the little things i commented on :)

---
In reply to: [277023975](https://github.com/Microsoft/sarif-sdk/pull/1439#discussion_r277023975) [](ancestors = 277023975)

Reply:
will do!

---
In reply to: [277024249](https://github.com/Microsoft/sarif-sdk/pull/1439#discussion_r277024249) [](ancestors = 277024249,277023975)